src/settings/slider-setting/slider-setting.ts

```diff
@@ -0,0 +1,70 @@
+import {UIPSetting} from '../setting/setting';
+import {attr} from '@exadel/esl/modules/esl-base-element/core';
+import {WARNING_MSG} from '../../utils/warning-msg';
+
+export class UIPSliderSetting extends UIPSetting {
+  public static is = 'uip-slider-setting';
+  /** CSS Class added when setting has inconsistent state. */
+  public static inconsistencyClass = 'inconsistency-marker';
+
+  /** Setting's visible name. */
+  @attr({defaultValue: ''}) public label: string;
+  /** Minimum range value. */
+  @attr({defaultValue: '0'}) public min: string;
+  /** Maximum range value. */
+  @attr({defaultValue: '100'}) public max: string;
+  /** Step for range. */
+  @attr({defaultValue: '1'}) public step: string;
+  /** Range input to change setting's value. */
+  protected $field: HTMLInputElement;
+  /** Range value displayed below. */
+  protected $fieldValue: HTMLDivElement;
+
+  protected connectedCallback() {
+    super.connectedCallback();
+    if (this.$field) return;
+
+    this.initSlider();
+    this.initSliderValue();
+    const label = document.createElement('label');
+    label.innerText = this.label;
+    label.appendChild(this.$field);
+
+    this.innerHTML = '';
+    this.append(label);
+    this.append(this.$fieldValue);
+  }
+
+  protected initSlider() {
```

Comment:
You can create memoized getter to simplify and optimize things.

Somewhere in the class: 
```
@memoize
protected get $field() {
   const $field = document.createElement('input');
   $field.type = 'range';
   return $field; 
} 

@memoize
protected get $label() {
   const $label = document.createElement('label');
   $label.appendChild(this.$field);
   return $label; 
} 

protected updateConfiguration() {
    this.$field.min = this.min;
    this.$field.max = this.max;
    this.$field.step = this.step;
    // Better wrap the text to the span to easily control it or render it trough ::before { content: attr(label); }
    // this.$label.innerText = this.label;
}

```

In Constructor:
```
this.updateConfiguration();
this.innerHTML = '';
this.append(this.$label);
this.append(this.$fieldValue);
```